### ISSUE TITLE
Skip kqueue reactor builder tests on Linux, where they will never ever run

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5623]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5623|
|Type|enhancement|
|Created|2012-04-08 00:37:41Z|
|Branch|https://github.com/twisted/twisted/tree/skip-kqueue-on-linux-5623|

Linux does not support kqueue.  Not reason to try to run these tests on Linux, and not even trying will reduce noise in the test output.

<details><summary>Searchable metadata</summary>

```
trac-id__5623 5623
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_skip_kqueue_on_linux_5623 branches/skip-kqueue-on-linux-5623
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1333845461000000 1333845461000000
changetime__1333914027000000 1333914027000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [34138]) Branching to 'skip-kqueue-on-linux-5623'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

_Pretty_ simple, just avoids adding kqueuereactor to the reactors list on Linux.  This if/elif tree is getting a bit complicated though.  Possibly the whole thing merits replacement by something a little nicer.  Not sure what, though.

Don't have time to force a build now, so someone else should do that when buildbot looks like it can handle it.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

I don't see a news file in the :12435 branch diff, but otherwise looks fine. Having a better framework at some point would be good, yes, especially if we can generalize skipping reactors that won't work, so maybe open a ticket for that after merging.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [34146]) Merge skip-kqueue-on-linux-5623 

Author: exarkun
Reviewer: itamar
Fixes: [#5623](https://github.com/twisted/twisted/issues/5623)

Avoid ever trying to run kqueue reactor tests on Linux, where kqueue is definitely
not supported.

This turns a bunch of explicit skips into a bunch of silent skips.
